server/cmd/mmctl/commands/post.go

```diff
@@ -103,6 +113,41 @@ func postCreateCmdF(c client.Client, cmd *cobra.Command, args []string) error {
 	return nil
 }
 
+func postCreateDMCmdF(c client.Client, cmd *cobra.Command, args []string) error {
+	ctx := context.TODO()
+	message := args[1]
+	if message == "" {
+		return errors.New("message cannot be empty")
+	}
+
+	otherUser, err := getUserFromArg(c, args[0])
+	if err != nil {
+		return fmt.Errorf("could not find user: %s", err.Error())
+	}
+
+	directChannel,_, err := c.CreateDirectChannel(ctx, otherUser.Id,"")
```

Comment:
@Kimbohlovette, you need to pass two userIDs. The userID of the user that runs the command is also required.

Reply:
The challenge I had at that point is that there is no direct way to get the id of the sender of the message so I set it to empty, and modified the implementation of the function handler CreateDirectChannel to fetch current user from session and use with the otherUserId

Reply:
There is a `GetMe` method that returns the current user. You can use that to fetch the user first and then call `CreateDirectChannel`.

Reply:
Alright thank you sir. It will help me to simplify many things

Reply:
Hello sir @hanzei 
When I used the GetMe method of the client it gave me the error below. When I try to inspect I did not see any endpoint at  web api4 with URL: /users/me. Could it be that it has not been implemented yet?

![Screenshot from 2024-09-27 18-19-33](https://github.com/user-attachments/assets/d964f102-d76c-40d3-a83b-7309aa4c1b65)

Reply:
If you are using the `--local` flag, there is no user associated with `mmctl`. You need to login first using `mmctl auth login` and then use the command without the `--local` flag.

Reply:
Okay sir I'll login and see

Reply:
Hello sir, it works now like a champ :laughing:

Reply:
Now that we have worked out the implemention to the end, could we simplify that command from
```
mmctl post create dm  <user>  '<message>'
```
to just
```
mmctl post dm <user> '<message>'
```
??

Reply:
Can we even extend the current `mmctl post create` command? Do we even need a separate one?

Reply:
Extending the `mmctl post create` is the ideal thing to do. But how do we differentiate between when someone is posting to a group and when they are posting to a user (DM)?

Reply:
Well, I will carefully observe the `postCreateCmdF` and see how I can extend it.

Reply:
Groups are a different problems. Users should be easily distinguished from channels. Just check if the user arguments starts with a `@`.

Reply:
Okay